feat: connect Postgres into Docker

K-Beicher opened the pull request:
## Description
This PR makes it possible to switch between a postgres database and a sqlite database.  The switch is done in the settings file under the section Databases.

The postgres database requires Docker Desktop to be up and running and the just command to run docker should be run prior to using 'just start-app' the first time.  Afterwards it should run as expected when the localhost is spun up.

## Related Issues

<!-- List issues the PR closes -->

Closes #373 
<!-- Connect this PR to relevant issues, to help the reviewer but also for record-keeping. -->

## Testing

- [ ] Yes
- [ ] No, not needed (give a reason below)
- [x] No, I need help writing them
I'm not sure how one would go about writing tests for this.

<!-- Please explain why the tests are not needed for this PR here -->

## Reviewer Focus
<!-- Please delete as appropriate: -->
This PR requires an in-depth review. 

There should be a check to see that localhost can still write metadata to the database and retain it.

If testing the db connection first make sure that the sqlight db is removed from your environment (use just reset-local), then run just start-docker, and just start-app.

<!-- Any particular section the reviewer should focus on, anywhere that would be a good place to start? -->

## Checklist

<!-- This is to help you determine if your work is ready to be reviewed, if an item is not relevant then you can mark it as done (because you have checked and found that it isn't needed) -->

For all PRs that are not general documentation

- [ ] Tests accompany or reflect changes to the code
- [x] Tests ran and passed locally
- [x] Ran the linter and formatter
- [x] Build has passed locally
- [ ] Relevant documentation has been updated

K-Beicher committed:
feat: first upload of docker compose file

K-Beicher committed:
feat: update dockerfile - remove 22 and 25

trying to get docker-compose up to run, playing around with Dockerfile

K-Beicher committed:
fixed typo

K-Beicher committed:
feat: update yml file to persist data in postgres

K-Beicher committed:
changed the dockerfile back

K-Beicher committed:
Merge branch 'main' into feat-connect-postgres

K-Beicher committed:
feat: remove linting issues

K-Beicher committed:
feat: update to run psycopg2-binary

K-Beicher committed:
Merge branch 'main' into feat-connect-postgres

philter87 commented on config/settings.py:
```diff
@@ -12,6 +12,7 @@
 import os
 from pathlib import Path
 
+# this import should be commented out if moving to postgres db
```
Remove this comment

philter87 commented on config/settings.py:
```diff
@@ -94,8 +95,21 @@
 sqlite_url = "sqlite:///db.sqlite3"
 database_url = os.environ.get("DATABASE_URL", sqlite_url)
 
+# The commented out db is the one we are running while using sqlite3
```
And remove this comment

philter87 commented on config/settings.py:
```diff
@@ -94,8 +95,21 @@
+# The commented out db is the one we are running while using sqlite3
 DATABASES = {"default": dj_database_url.parse(database_url, conn_max_age=600)}
-
+""" DATABASES = {
```
And remove this comment

philter87 reviewed (commented):
Yay! Very good. The only thing is that we forgot to remove some comments in the settings.py file which might confuse a bit

signekb commented:
We have tested this on my computer and even though we start docker, `sprout` still seems connected to `sqlite` instead of a postgres db. @K-Beicher will look into it ⚡

K-Beicher committed:
feat: update with comments by Philip

K-Beicher commented:
This should now be ready to test.  I updated based on comments that I had overlooked in the first instance.  Also added a shorthand for how to change over to the postgres db.

K-Beicher committed:
Merge branch 'main' into feat-connect-postgres

pchmia committed:
fix: .venv is added to .dockerignore as it seems to break the docker build

K-Beicher committed:
testing settings

K-Beicher committed:
Merge branch 'feat-connect-postgres' of https://github.com/seedcase-project/seedcase-sprout into feat-connect-postgres

K-Beicher committed:
expose 5432 in dockerfile

K-Beicher committed:
change host from localhost to 127.0.0.1

K-Beicher committed:
Merge branch 'main' into feat-connect-postgres

K-Beicher committed:
go from ' to " in docker-compose

K-Beicher committed:
Merge branch 'feat-connect-postgres' of https://github.com/seedcase-project/seedcase-sprout into feat-connect-postgres

K-Beicher committed:
remove host and port in settings

K-Beicher committed:
new database_url

K-Beicher committed:
try removing a database def

K-Beicher committed:
linting

K-Beicher committed:
put it all back

K-Beicher commented on docker-compose.yml:
```diff
@@ -1,10 +1,22 @@
+    image: 'postgres'
+    restart: always
+    ports:
+      - '5432:5432'
```
Sure, but it is failing at deployment, it is working locally.

K-Beicher committed:
remove second database setup in settings, change port

pchmia committed:
feat: small change

lwjohnst86 committed:
Merge branch 'main' into feat-connect-postgres

lwjohnst86 reviewed (approved):
We'll eventually need to actually connect the data import to go into the postgres backend, but this is a fantastic first step!! 🎉
